_posts/2021-12-13-hhast-4.139.0.markdown

```diff
@@ -0,0 +1,108 @@
+---
+title: "How to lint your code with the help of the type checker?"
+layout: post
+author: atry
+category: blog
+---
+
+HHAST is a toolkit for for analyzing the structure of Hack files. It includes a linting and migration framework to find and automatically fix problems in your code. The lint errors and migration suggestions can be accessed from Hack API, command line interface or IDEs.
+
+HHAST also ships with many built-in lint rules and migration rules. However, previously the HHAST built-in lint rules have limited access to the type checker, therefore, it is difficult to create a lint rule that needs to determine questions like "is this type a subtype of that type?" or "is this expression compatible with that function signature?".
```

Comment:
This is a bit misleading: we now make several available by default, but it's still difficult to create such rules - they need to be in ocaml, build in to the typechecker.

Reply:
Added a sentence to clarify that at https://github.com/hhvm/hhvm.com/pull/166/files#diff-e4b621d56854022ef11db693a074430ce03894df325c46da996a0bc4e33d9863R12